Formula/john.rb

```diff
@@ -14,45 +16,85 @@ class John < Formula
 
   conflicts_with "john-jumbo", :because => "both install the same binaries"
 
-  patch :DATA # Taken from MacPorts, tells john where to find runtime files
+  patch :DATA # Backporting official patch from jumbo fork (https://www.openwall.com/lists/john-users/2016/01/04/1)
```

Comment:
As this patch is quite long, could we put it in [Homebrew/formula-patches](https://github.com/Homebrew/formula-patches) and pull it in with a `patch do` block?

Reply:
Sure, but how do I go about doing this? The README there and the main docs say nothing about the required workflow to do so.

Reply:
Make a pull request to add it. That's all really

Reply:
https://github.com/Homebrew/formula-patches/pull/286 is merged now, so all that's left is to add `patch do` with the URL to the patch and its SHA256. There's [plenty of prior art](https://github.com/Homebrew/homebrew-core/search?q=Homebrew%2Fformula-patches&unscoped_q=Homebrew%2Fformula-patches).

Reply:
OK, done.